geoalchemy2/elements.py

```diff
@@ -131,11 +132,41 @@ class WKBElement(_SpatialElement):
 
     Note: you can create ``WKBElement`` objects from Shapely geometries
     using the :func:`geoalchemy2.shape.from_shape` function.
+
+
+    WKB struct {
+        byte    byteOrder;
+        uint32  wkbType;
+        uint32  SRID;
+        struct  geometry;
+    }
+
+    byteOrder enum {
+        WKB_XDR = 0,  // Most Significant Byte First
+        WKB_NDR = 1,  // Least Significant Byte First
+    }
+
     """
 
     geom_from = 'ST_GeomFromWKB'
     geom_from_extended_version = 'ST_GeomFromEWKB'
 
+    def __init__(self, data, srid=-1, extended=False):
+        if extended and srid == -1:  # unpack srid from WKB struct
+            header = data[:9]
+            try:  # support hex input
```

Comment:
Can you please remind me when we can get "hex input"? Is "hex input" something we already support? Or is it a new feature you are adding?

Reply:
Oh I guess we guess an hex string when using SpatiaLite.

Why not be consistent with the desc function and do this:

```py
if isinstance(data, str_):
    header = binascii.unhexlify(data[:18])
else:
    header = data[:9]
```

?

Reply:
It is not clear for me if hex input are supported (I find no mention in doc). You are right, I added it because of SpatiaLite tests.
I am not sure that checking data is instance of string is exact, because it seems to me that binary input could also be a string (not only hex input).
Maybe we need to change the desc function?

Reply:
> It is not clear for me if hex input are supported (I find no mention in doc). You are right, I added it because of SpatiaLite tests.

With SQLite/SpatiaLite we do get an hex string.

> I am not sure that checking data is instance of string is exact, because it seems to me that binary input could also be a string (not only hex input).

In the case of SpatiaLite we get a string, in the case of Postgis we get a memoryview (or a buffer in the case of Python 2).

So I think that the `isinstance(data, str_)` test that we use in `desc` is correct. I'd vote for consistency and using the same test in both `desc` and `__init__`.

Reply:
In the case of SpatiaLite and Python do we always get a **unicode** string?
Otherwise I also have to check if input is instance of `bytes`.

Reply:
In the case of SpatiaLite we get a unicode with Python 2 and a str with Python 3. This is why I am suggesting to use `isinstance(data, str_)`.

Reply:
ok :+1: